state-sync/storage-service/server/src/lib.rs

```diff
@@ -1244,6 +1321,63 @@ impl StorageReaderInterface for StorageReader {
         )))
     }
 
+    fn get_transactions_or_outputs_with_proof(
```

Comment:
FYI: this is the new logic. Everything else is boilerplate to handle the new request types 😄